pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -496,57 +402,13 @@ func (gce *GCECloud) EnsureLoadBalancerDeleted(clusterName string, service *v1.S
 	return nil
 }
 
-func (gce *GCECloud) DeleteForwardingRule(name string) error {
-	region, err := GetGCERegion(gce.localZone)
-	if err != nil {
-		return err
-	}
-	return gce.deleteForwardingRule(name, region)
-}
-
-func (gce *GCECloud) deleteForwardingRule(name, region string) error {
-	mc := newForwardingRuleMetricContext("delete", region)
-	op, err := gce.service.ForwardingRules.Delete(gce.projectID, region, name).Do()
-
-	if err != nil && isHTTPErrorCode(err, http.StatusNotFound) {
-		glog.Infof("Forwarding rule %s already deleted. Continuing to delete other resources.", name)
-	} else if err != nil {
-		glog.Warningf("Failed to delete forwarding rule %s: got error %s.", name, err.Error())
-		return mc.Observe(err)
-	} else {
-		if err := gce.waitForRegionOp(op, region, mc); err != nil {
-			glog.Warningf("Failed waiting for forwarding rule %s to be deleted: got error %s.",
-				name, err.Error())
-			return err
-		}
-	}
-	return nil
-}
-
-// DeleteTargetPool deletes the given target pool.
-func (gce *GCECloud) DeleteTargetPool(name string, hcNames ...string) error {
-	region, err := GetGCERegion(gce.localZone)
-	if err != nil {
-		return err
-	}
-	return gce.deleteTargetPool(name, region, hcNames...)
-}
-
-func (gce *GCECloud) deleteTargetPool(name, region string, hcNames ...string) error {
-	mc := newTargetPoolMetricContext("delete", region)
-	op, err := gce.service.TargetPools.Delete(gce.projectID, region, name).Do()
-
+func (gce *GCECloud) DeleteExternalTargetPoolAndChecks(name, region string, hcNames ...string) error {
```

Comment:
This was public before - I just renamed it and merged it with the logic of the non-public func.  https://github.com/kubernetes/kubernetes/pull/46663/files/b836aa75b780046cb290d3ab01f303ffb21ba316..10e59b429409a59c26f985f8a441626e169e37a0#diff-4cae6102def67324f04d6988b5aab0f6L527

Unfortunately, it's public because it's being used in resource cleanup in `test/e2e/framework/utils.go`.